src/openvic-simulation/modifier/ModifierManager.cpp

```diff
@@ -686,181 +661,140 @@ bool ModifierManager::parse_scripts(DefinitionManager const& definition_manager)
+NodeTools::key_value_callback_t ModifierManager::_expect_modifier_effect(
+	modifier_effect_registry_t const& registry,
+	ModifierValue& modifier_value
 ) const {
-	return expect_validated_modifier_value_and_default(modifier_callback, type, key_value_invalid_callback, effect_validator);
+	return _expect_modifier_effect_with_fallback(registry, modifier_value, key_value_invalid_callback);
 }
-
-node_callback_t ModifierManager::expect_modifier_value_and_default(
-	callback_t<ModifierValue&&> modifier_callback, Modifier::modifier_type_t type, key_value_callback_t default_callback
+NodeTools::key_value_callback_t ModifierManager::_expect_modifier_effect_with_fallback(
+	modifier_effect_registry_t const& registry,
+	ModifierValue& modifier_value,
+	const NodeTools::key_value_callback_t fallback
 ) const {
-	return expect_validated_modifier_value_and_default(
-		modifier_callback, type, default_callback, [](ModifierEffect const&) -> bool {
-			return true;
+	return [this, &registry, &modifier_value, fallback]
+		(const std::string_view key, const ast::NodeCPtr value) {
+		if (dryad::node_has_kind<ast::ListValue>(value) && complex_modifiers.contains(key)) {
+			return expect_dictionary([this, &modifier_value, &key]
+				(const std::string_view inner_key, const ast::NodeCPtr inner_value) -> bool{
+				return _add_flattened_modifier_cb(modifier_value, key, inner_key, inner_value);
+			})(value);
 		}
-	);
+	
+		ModifierEffect const* effect = registry.get_item_by_identifier(key);
+		if (effect == nullptr) {
+			return fallback(key, value);
+		}
+		return _add_modifier_cb(modifier_value, effect, value);
+	};
 }
-
-node_callback_t ModifierManager::expect_modifier_value(
-	callback_t<ModifierValue&&> modifier_callback, Modifier::modifier_type_t type
-) const {
-	return expect_modifier_value_and_default(modifier_callback, type, key_value_invalid_callback);
+NodeTools::key_value_callback_t ModifierManager::_expect_shared_tech_country_modifier_effect(ModifierValue& modifier_value) const {
+	return _expect_modifier_effect(shared_tech_country_modifier_effects, modifier_value);
 }
-
-node_callback_t ModifierManager::expect_whitelisted_modifier_value_and_default(
-	callback_t<ModifierValue&&> modifier_callback, Modifier::modifier_type_t type, string_set_t const& whitelist,
-	key_value_callback_t default_callback
-) const {
-	return expect_validated_modifier_value_and_default(
-		modifier_callback, type, default_callback,
-		[&whitelist](ModifierEffect const& effect) -> bool {
-			return whitelist.contains(effect.get_identifier());
-		}
-	);
+NodeTools::key_value_callback_t ModifierManager::expect_leader_modifier(ModifierValue& modifier_value) const {
+	return _expect_modifier_effect(leader_modifier_effects, modifier_value);
 }
+NodeTools::key_value_callback_t ModifierManager::expect_technology_modifier(ModifierValue& modifier_value) const {
+	return [this, &modifier_value](const std::string_view key, const ast::NodeCPtr value) {
```

Comment:
```suggestion
	return [this, &modifier_value](const std::string_view key, const ast::NodeCPtr value) -> bool {
```
Explicit return type for lambda function, add this to any others missing it too (e.g. in `expect_province_event_modifier`)

Reply:
Ugh it feels like I'm programming in nodepad, does the IDE do nothing for me?